delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/ResourceIdCheck.scala

```diff
@@ -0,0 +1,49 @@
+package ch.epfl.bluebrain.nexus.delta.sdk
+
+import ch.epfl.bluebrain.nexus.delta.kernel.RetryStrategy
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.UrlUtils
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.DBModuleType
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.persistenceid.PersistenceIdCheck
+import com.typesafe.scalalogging.Logger
+import monix.bio.IO
+import retry.syntax.all._
+
+class ResourceIdCheck(
+    idCheck: PersistenceIdCheck,
+    dbModuleTypes: Set[DBModuleType],
+    retryStrategy: RetryStrategy[Throwable]
+) {
+
+  private val logger: Logger = Logger[ResourceIdCheck]
+
+  /**
+    * Checks whether an id is available (does not already exist) across different resource types
+    *
+    * @param project  the project
+    * @param id       the id
+    * @param onExists the value to return in the error channel when the resource already exists
+    * @tparam R the error type
+    * @return () on the regular channel if the passed id does not already exist. Return ''onExists'' on the error channel otherwise
+    */
+  def isAvailable[R](project: ProjectRef, id: Iri, onExists: => R): IO[R, Unit] = {
+    val persistenceIds = dbModuleTypes.map { moduleType =>
+      s"${moduleType.value}-${UrlUtils.encode(project.toString)}_${UrlUtils.encode(id.toString)}"
+    }
+    idCheck
+      .existsAny(persistenceIds.toSeq: _*)
+      .retryingOnSomeErrors(retryStrategy.retryWhen, retryStrategy.policy, retryStrategy.onError)
+      .redeemWith(
```

Comment:
I don't see why this is wrong. The error type and message after retries is misleading yes, but I think it is a good compromise. We cannot propagate the error (as a Task) to the upper layer because the evaluate methods require an `IO[E, A]` where E is not a subtype of Throwable. 

Without changing a lot of the code I don't see how this could be easily done (expect for `hideError` which is probably worse than the current solution)

Reply:
Everytime we fetch a resource when evaluating, we already do a `hideErrors` so we should be consistent and `hideErrors` is fine for the moment.
We can think about a better way to handle these errors post 1.5.

Reply:
Ok. I can hideErrors but I would still keep the retry, so that at least we have a change to recover from some timeout